crates/dojo-core/src/world/world_contract.cairo

```diff
@@ -734,12 +709,17 @@ pub mod world {
             ref self: ContractState, selector: felt252, class_hash: ClassHash
         ) -> ClassHash {
             assert(self.is_account_owner(selector), Errors::NOT_OWNER);
-            let (_, contract_address) = self.contract(selector);
-            IUpgradeableDispatcher { contract_address }.upgrade(class_hash);
-            EventEmitter::emit(
-                ref self, ContractUpgraded { class_hash, address: contract_address }
-            );
-            class_hash
+
+            if let Resource::Contract((_, contract_address)) = self.resources.read(selector) {
+                IUpgradeableDispatcher { contract_address }.upgrade(class_hash);
+                EventEmitter::emit(
+                    ref self, ContractUpgraded { class_hash, address: contract_address }
+                );
+                class_hash
+            } else {
+                core::panic_with_felt252(Errors::INVALID_RESOURCE_SELECTOR);
+                core::num::traits::Zero::<ClassHash>::zero()
```

Comment:
We don't need to return anything if we panic. If you remove the `;` after the panic the compiler should not complain of different types in the `if let` branch.

Reply:
Ha ! That's the trick ! Thanks 👍